src/sentry/workflow_engine/models/detector.py

```diff
@@ -105,3 +103,11 @@ def detector_handler(self) -> DetectorHandler | None:
     def get_audit_log_data(self) -> dict[str, Any]:
         # TODO: Create proper audit log data for the detector, group and conditions
         return {}
+
+
+@receiver(pre_save, sender=Detector)
+def enforce_config_schema(sender, instance: Detector, **kwargs):
+    from sentry.workflow_engine.registry import detector_config_schema_registry
+
+    config_schema = detector_config_schema_registry.get(instance.type)
```

Comment:
Do we actually need the registry, or since we have the detector, should we use `instance.group_type.detector_config_schema`?